textattack/shared/scripts/run_attack_single_threaded.py

```diff
@@ -27,13 +28,18 @@ def run(args):
         # Override current args with checkpoint args
         resume_checkpoint = parse_checkpoint_from_args(args)
         args = merge_checkpoint_args(resume_checkpoint.args, args)
-        num_examples_offset = resume_checkpoint.dataset_offset
-        num_examples = resume_checkpoint.num_remaining_attacks
+        
+        num_remaining_attacks = resume_checkpoint.num_remaining_attacks
+        worklist = resume_checkpoint.worklist.copy()
+        assert num_remaining_attacks == len(worklist), "Recorded number of remaining attacks and size of worklist are different."
+        last_example = resume_checkpoint.last_example
+
         logger.info('Recovered from checkpoint previously saved at {}'.format(resume_checkpoint.datetime))
         print(resume_checkpoint, '\n')
     else:
-        num_examples_offset = args.num_examples_offset
-        num_examples = args.num_examples
+        num_remaining_attacks = args.num_examples
+        worklist = deque(range(0, args.num_examples))
+        last_example = worklist[-1]
```

Comment:
I think I understand what `last_example` is doing, but can you rename it (maybe to `last_example_idx` or something else clearer) and add a comment or two clarifying that it's necessary to enforce `--attack-n`?

Reply:
Yeah it needs a better name.